src/types.ts

```diff
@@ -96,16 +96,36 @@ export interface Instance {
   ): Promise<ethers.ContractTransaction>;
 
   /**
-   * Update the metadata of a given domain
-   * @param domainId The domain with metadata to be locked/unlocked
-   * @param metadataUri The link that the metadata should be updated to
-   * @param signer The account that signs and sends the transaction
+   * Get the metadata uri for a given domain
+   * 
+   * @param domainId The domain to get metadata for
+   * @param signer The account used in instantiating the registrar
    */
-  setDomainMetadata(
+  getDomainMetadata: (
     domainId: string,
-    metadataUri: string,
     signer: ethers.Signer
-  ): Promise<ethers.ContractTransaction>;
+  ) => Promise<DomainMetadata>;
+  /**
+   * Set the domain metadata to a given metadata object
+   * 
+   * @param domainId The id of the domain to set
+   * @param metadata An object containing any desired properties of the updated metadata
+   * @param signer: The account used in the transaction, which must be the owner of the domain 
+   */
+  setDomainMetadata: (
```

Comment:
Nit: Seems like we have 2 functions for setting metadata but they work differently:
- `setDomainMetadata` takes a `DomainMetadata` object as input
- `setAndLockDomainMetadataUri` takes a `string` uri as input

It would be better if we had a consistent interface for both. Could we create two more functions `setDomainMetadataUri` (takes string uri) and `setAndLockDomainMetadata` (takes `DomainMetadata` object)?